packages/frameworks-sveltekit/src/lib/index.ts

```diff
@@ -251,19 +270,19 @@ function AuthHandle(prefix: string, authOptions: AuthConfig): Handle {
       return resolve(event)
     }
 
-    return Auth(request, authOptions)
+    return Auth(request, resolvedAuthOptions)
   }
 }
 
 /**
  * The main entry point to `@auth/sveltekit`
  * @see https://sveltekit.authjs.dev
  */
-export function SvelteKitAuth(options: SvelteKitAuthConfig): Handle {
+export function SvelteKitAuth(options: SvelteKitAuthConfig, dynamicOptions?: OptionsFunction): Handle {
   const { prefix = "/auth", ...authOptions } = options
   authOptions.secret ??= env.AUTH_SECRET
   authOptions.trustHost ??= !!(env.AUTH_TRUST_HOST ?? env.VERCEL ?? dev)
-  return AuthHandle(prefix, authOptions)
+  return AuthHandle(prefix, authOptions, dynamicOptions)
```

Comment:
IMO we should only keep one parameter for `options`, so it could be typed as `SvelteKitAuthConfig | DynamicSvelteKitAuthConfig`

Let's do it like this:
```ts
export function SvelteKitAuth(
  options: SvelteKitAuthConfig | DynamicSvelteKitAuthConfig
): Handle {
  if (typeof options === "object") {
    options.secret ??= env.AUTH_SECRET
    options.trustHost ??= !!(env.AUTH_TRUST_HOST ?? env.VERCEL ?? dev)
  }
  return AuthHandle(options)
}
```

Since the assignment of `secret` and `trustHost` requires `env`, we have to do it in the function callback anyway.
To keep the `prefix` options working let's move its default value assignment part to inside `AuthHandle`

Reply:
Sounds good.  I like it, it's clean.  I pushed a commit that keeps the api signature, keeps the default /auth prefix for both static and dynamic options and keeps getting values for secret and TrustHost from env in the static configuration.   I have done some basic end to end testing locally and it works in my environment.   I think it achieves what you want.